src/ResourceManagement/HDInsightJob/HDInsightJob.Tests/ScenarioTests/SubmitJobTests.cs

```diff
@@ -544,31 +736,43 @@ public void SubmitHiveJobError()
             }
         }
 
-        private static string Convert(Stream stream)
-        {
-            var reader = new StreamReader(stream);
-            var text = reader.ReadToEnd();
-            return text;
-        }
-
-        private JobGetResponse GetJobFinalStatus(HDInsightJobManagementClient client, string jobId)
+        internal void WaitForJobCompletion(HDInsightJobManagementClient client, string jobId, TimeSpan pollingInterval, TimeSpan duration, bool cancelJob = false)
         {
-            var jobStatus = client.JobManagement.GetJob(jobId);
+            var startTime = DateTime.UtcNow;
+            var endTime = DateTime.UtcNow;
 
-            while (!jobStatus.JobDetail.Status.JobComplete)
+            var jobDetail = client.JobManagement.GetJob(jobId);
+            while (!jobDetail.JobDetail.Status.JobComplete)
             {
-                jobStatus = client.JobManagement.GetJob(jobId);
+                if (((endTime = DateTime.UtcNow) - startTime) > duration)
+                {
+                    string exceptionMessage = string.Format(CultureInfo.InvariantCulture, "The requested task failed to complete in the allotted time ({0})", duration); ;
+                    if (cancelJob)
+                    {
+                        client.JobManagement.KillJob(jobId);
+                        exceptionMessage = string.Format(CultureInfo.InvariantCulture, "{0} Killed the Job {1}", exceptionMessage, jobId);
+                    }
+
+                    throw new CloudException(exceptionMessage);
+                }
 
-                // Optional to sleep here. Sleep for 1 sec instead of keep pooling. 
-                Thread.Sleep(1000);
+                Thread.Sleep(pollingInterval);
```

Comment:
@pattipaka you should not use Thread.Sleep. Use TestMockSupport.Delay method instead.